pg_replicate/src/clients/delta.rs

```diff
@@ -0,0 +1,235 @@
+            _ => DataType::STRING,
+        }
+    }
+
+    fn postgres_to_arrow(typ: &Type) -> ArrowDataType {
+        match typ {
+            &Type::BOOL => ArrowDataType::Boolean,
+            &Type::CHAR | &Type::BPCHAR | &Type::VARCHAR | &Type::NAME | &Type::TEXT => {
+                ArrowDataType::Utf8
+            }
+            &Type::INT2 | &Type::INT4 | &Type::INT8 => ArrowDataType::Int32,
+            &Type::FLOAT4 | &Type::FLOAT8 | &Type::NUMERIC => ArrowDataType::Float16,
+            &Type::DATE => ArrowDataType::Date32,
+            &Type::TIME | &Type::TIMESTAMP | &Type::TIMESTAMPTZ => ArrowDataType::Utf8,
+            &Type::UUID => ArrowDataType::Utf8,
+            &Type::OID => ArrowDataType::Utf8,
+            _ => ArrowDataType::Utf8,
+        }
+    }
+
+    pub async fn create_table_schema(
+        &mut self,
+        tb_name: String,
+        columns: &[ColumnSchema],
+    ) -> Result<Arc<Schema>, DeltaTableError> {
+        let full_path = self.delta_full_path(tb_name.clone());
+
+        if Path::new(&full_path).join("_delta_log").exists() {
+            DeltaOps::try_from_uri(&full_path).await?.delete().await?;
+        }
+
+        let mut table = DeltaOps::try_from_uri(&full_path)
+            .await?
+            .create()
+            .with_table_name(tb_name.clone());
+
+        let arrow_schema = Self::generate_schema(columns, &mut table)?;
+
+        Ok(arrow_schema)
+    }
+
+    fn generate_schema(
+        columns: &[ColumnSchema],
+        table: &mut CreateBuilder,
+    ) -> Result<Arc<Schema>, DeltaTableError> {
+        let mut schema: Vec<Field> = vec![];
+
+        for column in columns {
+            *table = table.clone().with_column(
+                column.name.as_str(),
+                Self::postgres_to_delta(&column.typ),
+                false,
+                None,
+            );
+
+            schema.push(Field::new(
+                column.name.as_str(),
+                Self::postgres_to_arrow(&column.typ),
+                true,
+            ));
+        }
+
+        schema.push(Field::new("OP", ArrowDataType::Utf8, true));
+        schema.push(Field::new(
+            "pg_replicate_inserted_time",
+            ArrowDataType::Timestamp(TimeUnit::Microsecond, None),
+            true,
+        ));
+
+        Ok(Arc::new(Schema::new(schema)))
+    }
+
+    fn get_table_schema(&self, table_id: TableId) -> Result<&TableSchema, DeltaTableError> {
+        self.table_schemas
+            .as_ref()
+            .ok_or_else(|| {
+                DeltaTableError::Generic("Table schemas are not initialized".to_string())
+            })?
+            .get(&table_id)
+            .ok_or_else(|| {
+                DeltaTableError::Generic(format!("Table schema not found for ID: {:?}", table_id))
+            })
+    }
+
+    fn get_delta_schema(&self, table_name: String) -> Result<&Arc<Schema>, DeltaTableError> {
+        self.delta_schemas
+            .as_ref()
+            .ok_or_else(|| {
```

Comment:
These will probably be removed if delta_schemas becomes a `HashMap` instead of `Option<HashMap>`.

Reply:
need Option<HashMap>